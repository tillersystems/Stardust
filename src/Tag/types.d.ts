import { ReactNode } from 'react';
import { DefaultTheme } from 'styled-components';

/**
 * Typescript interface.
 */
export interface TagProps {
  /** Anything that can be rendered: numbers, strings, elements or an array (or fragment */
  children: ReactNode;
  color?: string;
  theme: DefaultTheme;
}
