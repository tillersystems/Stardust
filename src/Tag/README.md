# Tag

### Usage

```jsx
import { Tag } from '@tillersystems/stardust';
```

<!-- STORY -->

### Properties

- `children` - anything that can be rendered: numbers, strings, elements or an array (or fragment).
- `className` - className needed by styled components.
- `color` - the color of tag box.

| propName    | propType | defaultValue | isRequired |
| ----------- | :------: | :----------: | :--------: |
| `children`  |  `node`  |    `null`    |     -      |
| `className` | `string` |     `''`     |     -      |
| `color`     | `string` |    `` | -    |

### Example

```jsx
import { Tag } from '@tillersystems/stardust';

render() {
  return <Tag color="red">Error</CheckBox>
}
```
