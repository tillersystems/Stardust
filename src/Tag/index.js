import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * A Tag categorizes or marks anything.
 *
 * @return {jsx}
 */

const Tag = ({ children, className, color, ...rest }) => (
  <div className={className} {...rest} data-testid="tag">
    {children}
  </div>
);

/**
 * PropTypes Validation.
 */
const { node, string } = PropTypes;
Tag.propTypes = {
  /** Anything that can be rendered: numbers, strings, elements or an array (or fragment) */
  children: node,

  /** className needed by styled-components */
  className: string,

  /** The background color of the tag box */
  color: string,
};

/**
 * Default props.
 */
Tag.defaultProps = {
  children: null,
  className: '',
  color: '',
};

export default styled(Tag)`
  position: relative;
  display: inline-block;

  padding: 0.2rem 0.8rem 0.4rem;
  border-radius: 0.5rem;

  color: ${({ theme: { palette } }) => palette.white};
  background: ${({ color }) => color};
`;
