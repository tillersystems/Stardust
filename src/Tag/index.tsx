import styled from 'styled-components';

import { TagProps } from './types.d';
import Theme from '../Theme';

/**
 * A Tag categorizes or marks anything.
 */
const Tag = styled.div<TagProps>(({ theme: { palette }, color }) => {
  return `
  position: relative;
  display: inline-block;

  padding: 0.4rem 0.8rem;
  border-radius: 0.5rem;

  color: ${palette.white};
  background: ${color};
`;
});

/**
 * Default props.
 */
Tag.defaultProps = {
  color: Theme.palette.primary.default,
} as Partial<TagProps>;

export default Tag;
