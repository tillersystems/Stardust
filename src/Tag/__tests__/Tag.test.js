import React from 'react';
import 'jest-styled-components';

import Tag from '..';

describe('<Tag />', () => {
  test('should render without a problem', () => {
    const { container } = render(<Tag>Beta</Tag>);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should have a color', () => {
    const { getByText } = render(<Tag color="red">Beta</Tag>);

    expect(getByText('Beta')).toHaveStyleRule('background', 'red');
  });
});
