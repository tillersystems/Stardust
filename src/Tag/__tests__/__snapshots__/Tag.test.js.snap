// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should have a color 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  padding: 0.2rem 0.8rem 0.4rem;
  border-radius: 0.5rem;
  color: hsl(0,100%,100%);
  background: red;
}

<div
  className="c0"
/>
`;

exports[`<Tag /> should render withouth a problem 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  padding: 0.2rem 0.8rem 0.4rem;
  border-radius: 0.5rem;
  color: hsl(0,100%,100%);
}

<div
  className="c0"
/>
`;
